### PR TITLE
ras_extended.py: Added test to verify usysident functionality

### PR DESCRIPTION
**Pull request by shirishaganta1**
Checking to turn on device identify indicators and other help
options of usysident

Signed-off-by: Shirisha Ganta <SHIRISHA.Ganta1@ibm.com>

### CODE REVIEW

**Comment by shirishaganta1** [comment]
After Patch
\# avocado run --test-runner runner ras_extended.py 
JOB ID     : 9c514ea1ff7d14a8bea6c18409aa0434ea3e04d1
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-06-21T21.01-9c514ea/job.log
 (01/12) ras_extended.py:RASTools.test1_uesensor: PASS (1.32 s)
 (02/12) ras_extended.py:RASTools.test1_serv_config: PASS (1.37 s)
 (03/12) ras_extended.py:RASTools.test1_ls_vscsi: PASS (1.29 s)
 (04/12) ras_extended.py:RASTools.test1_ls_veth: PASS (1.33 s)
 (05/12) ras_extended.py:RASTools.test1_ls_vdev: PASS (1.38 s)
 (06/12) ras_extended.py:RASTools.test1_lsdevinfo: PASS (1.82 s)
 (07/12) ras_extended.py:RASTools.test1_hvcsadmin: PASS (1.45 s)
 (08/12) ras_extended.py:RASTools.test1_bootlist: PASS (2.00 s)
 (09/12) ras_extended.py:RASTools.test1_vpdupdate: PASS (2.43 s)
 (10/12) ras_extended.py:RASTools.test3_lsvpd: PASS (3.18 s)
 (11/12) ras_extended.py:RASTools.test3_lscfg: PASS (1.79 s)
 (12/12) ras_extended.py:RASTools.test_usysident: PASS (9.53 s)
RESULTS    : PASS 12 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-06-21T21.01-9c514ea/results.html
JOB TIME   : 47.96 s

**Review by PraveenPenguin — APPROVED** [review]

File: ras/ras_extended.py [hunk 1]
```diff
@@ -249,3 +249,18 @@ def test3_lscfg(self):
         if self.is_fail >= 1:
             self.fail("%s command(s) failed in lscfg tool verification"
                       % self.is_fail)
+
+    def test_usysident(self):
+        self.log.info("============Executing usysident tool test============")
+        list = ['-h', '-V', '-P']
```
Comment (PraveenPenguin) [inline ras/ras_extended.py:255]: can we drive this from YAML as well

File: ras/ras_extended.py [hunk 2]
```diff
@@ -249,3 +249,18 @@ def test3_lscfg(self):
         if self.is_fail >= 1:
             self.fail("%s command(s) failed in lscfg tool verification"
                       % self.is_fail)
+
+    def test_usysident(self):
```
Comment (Naresh-ibm) [inline ras/ras_extended.py:253]: @shirishaganta1 please add the docstring for this function.

**Comment by Naresh-ibm** [comment]
@shirishaganta1 also please attach the job.log file.

File: ras/ras_extended.py (hunk 1 above)
Comment (shirishaganta1) [reply ras/ras_extended.py]: okay will do that

**Comment by shirishaganta1** [comment]
# avocado run --test-runner runner ras_extended.py -m ras_extended.py.data/options.yaml 
JOB ID     : 99f96b7dfcc45be772c66dbfe0f3714294b8960d
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-06-23T22.45-99f96b7/job.log
 (01/13) ras_extended.py:RASTools.test1_uesensor;run-b910: PASS (1.43 s)
 (02/13) ras_extended.py:RASTools.test1_serv_config;run-b910: PASS (1.65 s)
 (03/13) ras_extended.py:RASTools.test1_ls_vscsi;run-b910: PASS (1.53 s)
 (04/13) ras_extended.py:RASTools.test1_ls_veth;run-b910: PASS (1.57 s)
 (05/13) ras_extended.py:RASTools.test1_ls_vdev;run-b910: PASS (1.58 s)
 (06/13) ras_extended.py:RASTools.test1_lsdevinfo;run-b910: PASS (2.86 s)
 (07/13) ras_extended.py:RASTools.test1_hvcsadmin;run-b910: PASS (1.90 s)
 (08/13) ras_extended.py:RASTools.test1_bootlist;run-b910: PASS (2.65 s)
 (09/13) ras_extended.py:RASTools.test1_vpdupdate;run-b910: PASS (2.74 s)
 (10/13) ras_extended.py:RASTools.test3_lsvpd;run-b910: PASS (2.85 s)
 (11/13) ras_extended.py:RASTools.test3_lscfg;run-b910: PASS (1.99 s)
 (12/13) ras_extended.py:RASTools.test_usysident;run-b910: PASS (4.27 s)
 (13/13) ras_extended.py:RASTools.test_locking_mechanism;run-b910: FAIL: Database corruption detected (17.25 s)
RESULTS    : PASS 12 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-06-23T22.45-99f96b7/results.html
JOB TIME   : 47.50 s
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/8973015/job.log)

File: ras/ras_extended.py [hunk 3]
```diff
@@ -249,3 +249,22 @@ def test3_lscfg(self):
...
+        """
+        This tests to turn on device identify indicators and other help
+        options of usysident
+        """
+        self.log.info("============Executing usysident tool test============")
```
Comment (abdhaleegit) [inline ras/ras_extended.py:258]: remove the log info, not needed

File: ras/ras_extended.py [hunk 4]
```diff
@@ -249,3 +249,22 @@ def test3_lscfg(self):
...
+        value = self.params.get('usysident_list', default=['-h', '-V', '-P'])
+        for list_item in value:
+            if 'not supported' in self.run_cmd_out('usysident  %s ' % list_item):
+                self.cancel(
+                    "The identify indicators are not supported on this system")
```
Comment (abdhaleegit) [inline ras/ras_extended.py:263]: what if -P is supported options, if -V is not supported, the whole test gets cancled before testing -P ?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

**Comment by PraveenPenguin** [comment]
@shirishaganta1 can you please check/address the review comments

**Comment by shirishaganta1** [comment]
Addressed all the above comments and pushed the changes

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**